addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -554,6 +554,20 @@ class MessageList extends Component {
         this._loadMore();
     }
 
+    /**
+     * @private
+     */
+    _onClickRetryLoadMoreMessages() {
+        if (!this.threadView) {
+            return;
+        }
+        if (!this.threadView.threadCache) {
+            return;
+        }
+        this.threadView.threadCache.update({ hasLoadingFailed: false });
+        this._loadMore();
```

Comment:
**Comment here doesn't request changes, you can keep like this for this PR. Just to know we will make some cleaning here and there because there are some flaws in the design.**

______

Not super found of the design with side-effect on `hasToLoadMessages`. It's not obvious here that setting this field will actually load messages again. And it seems off for component to change internal state of a model itself...

It's weird having both the explicit `_loadMore` and the side-effect `hasToLoadMessages` happen at the same time.

I get the `hasToLoadMessages` is about 1st load, not the load more flow, so that's why there's not actually conflicting behaviour here. I'm just pointing out the strange design.

I think this problem will solve itself with the plan for future changes, starting with my PR to technically clean models, their fields, and how components observe changes in models. Following cleaning will consist in removing the inappropriate side-effects such as computed method `onChangeX`, which most of them are quite bad.